### PR TITLE
2020-09-01-preview Batch endpoints and deployments

### PR DESCRIPTION
**Pull request by puviyarasan**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.

**Comment by puviyarasan** [comment]
> While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.

As replied over email, "There are no existing APIs that can create batch inference endpoints / deployment resources today, these are new APIs that will be added in. All the JSONs included in this PR will be new files, in the preview version, let me know if I’m missing anything."

**Comment by puviyarasan** [comment]
@chiragg4u/ @markcowl, Can someone add me as a member to the [open-api-portal team in GH ](https://github.com/orgs/Azure/teams/openapi-portal-users/members?query=)? I have a pending request for several days and no response yet. At the moment, I'm not able to add labels/ reply to any comments.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json [hunk 1]
```diff
@@ -0,0 +1,1004 @@
...
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchDeploymentResource"
+            }
+          },
+          "201": {
```
Comment (chiragg4u) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:98]: >201 [](start = 11, length = 3)

Please add long x-ms-longrunning

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json [hunk 2]
```diff
@@ -0,0 +1,1004 @@
...
+      "description": "Batch inference settings per deployment.",
+      "type": "object",
+      "properties": {
+        "scaleSettings": {
+          "$ref": "#/definitions/DeploymentScaleSettings"
+        },
+        "batchSettings": {
+          "$ref": "#/definitions/BatchDeploymentSettings"
+        },
+        "tags": {
```
Comment (chiragg4u) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:824]: >tags [](start = 9, length = 4)

Looks like there are multiple level of tags. Can you explain the usage of these tags?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json [hunk 3]
```diff
@@ -0,0 +1,1004 @@
...
+            "create",
+            "read",
+            "update"
+          ]
+        },
+        "description": {
+          "description": "Description of the endpoint deployment.",
+          "type": "string"
+        },
+        "properties": {
```
Comment (chiragg4u) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:840]: >properties [](start = 9, length = 10)

Same here there are multi level of properties. What is the purpose?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json [hunk 4]
```diff
@@ -0,0 +1,1004 @@
...
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchEndpointResource"
+            }
+          },
+          "201": {
```
Comment (chiragg4u) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:320]: >201 [](start = 11, length = 3)

Add x-ms-longrunning

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json (hunk 3 above)
Comment (puviyarasan) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:840]: There is only 1 level of properties here, the external 'properties' is the resource envelope wrapper?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json (hunk 2 above)
Comment (puviyarasan) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:824]: The inner tags are redundant, removing them.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json (hunk 1 above)
Comment (puviyarasan) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:98]: Adding it

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json (hunk 4 above)
Comment (puviyarasan) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json:320]: Adding it..

**Review by chiragg4u — APPROVED** [review]
:shipit:

**Comment by markcowl** [comment]
@puviyarasan Please take a look at the python SDK changes - are these expected?

**Pull request merged**